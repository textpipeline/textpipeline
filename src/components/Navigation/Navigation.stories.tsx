import { storiesOf } from '@storybook/react';
import React from 'react';

import Navigation from './Navigation';

storiesOf('Navigation', module).add('default', () => <Navigation width={240} />);
