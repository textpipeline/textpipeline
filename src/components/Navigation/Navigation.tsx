import Box from '@material-ui/core/Box';
import Divider from '@material-ui/core/Divider';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import EditIcon from '@material-ui/icons/Edit';
import React from 'react';

import Heading from '../Heading';

export interface NavigationProps {
  readonly siteName: string;
}

const Navigation: React.FC<NavigationProps> = ({ siteName }) => (
  <>
    <Box p={2}>
      <Heading level={1} align="center">
        {siteName}
      </Heading>
    </Box>
    <Divider />
    <List>
      {['CSV to XLS', 'XML to JSON', 'Typescript to Javascript', 'SCSS to CSS'].map(text => (
        <ListItem button key={text}>
          <ListItemIcon>
            <EditIcon />
          </ListItemIcon>
          <ListItemText primary={text} />
        </ListItem>
      ))}
    </List>
  </>
);

export default Navigation;
