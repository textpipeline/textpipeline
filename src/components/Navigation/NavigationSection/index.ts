export { default } from './NavigationSection';
export * from './NavigationSection';
