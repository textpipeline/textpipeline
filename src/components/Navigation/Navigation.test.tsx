import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import Navigation from './Navigation';

describe('Navigation', () => {
  let rendered: ReturnType<typeof mount>;

  describe('when visited from a known path', () => {
    beforeEach(() => {
      rendered = mount(
        <MemoryRouter initialEntries={[{ pathname: '/', key: 'test' }]}>
          <Navigation siteName="Test Site" />
        </MemoryRouter>
      );
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('when visited from an unknown path', () => {
    beforeEach(() => {
      rendered = mount(
        <MemoryRouter initialEntries={[{ pathname: '/fnarglefloof', key: 'test' }]}>
          <Navigation siteName="Test Site" />
        </MemoryRouter>
      );
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });
});
