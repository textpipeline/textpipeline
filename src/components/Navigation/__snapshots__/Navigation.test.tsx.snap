// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<WithStyles(ForwardRef(Drawer))
  anchor="left"
  className="makeStyles-drawer-1 makeStyles-drawer-3"
  classes={
    Object {
      "paper": "makeStyles-drawerPaper-2 makeStyles-drawerPaper-4",
    }
  }
  variant="permanent"
>
  <Header />
  <NavigationSection
    isPrimary={true}
  >
    <TransformList />
  </NavigationSection>
  <NavigationSection>
    <Footer />
  </NavigationSection>
</WithStyles(ForwardRef(Drawer))>
`;
